packages/mds-ingest-service/service/provider.ts

```diff
@@ -189,13 +215,17 @@ export const IngestServiceProvider: ServiceProvider<IngestService & IngestMigrat
     }
 
     try {
-      await IngestRepository.writeMigratedTelemetry(
-        [eventTelemetryModel(telemetry, { recorded: event.recorded })],
+      const migratedTelemetry = await writeMigratedTelemetry(
+        eventTelemetryModel(telemetry, { recorded: event.recorded }),
         migrated_from
       )
-      const [migrated = null] = await IngestRepository.writeMigratedVehicleEvent([event], migrated_from)
-      if (migrated) {
-        const model = { ...migrated, telemetry: eventTelemetryModel(telemetry, { recorded: event.recorded }) }
+      if (!migratedTelemetry) {
+        return ServiceResult(migratedTelemetry) // should basically never happen
+      }
```

Comment:
This should not return here. Need to write event even if telemetry already existed.

Reply:
Fixed in c670dba